nixos/modules/installer/tools/nixos-generate-config/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+use std::error;
+use std::io;
+use std::io::Write;
+use std::process::Command;
+
+use simple_error::{bail, try_with};
+
+const MAN_EXECUTABLE: &str = env!("MAN_EXECUTABLE");
+const NIXOS_MANPAGES: &str = env!("NIXOS_MANPAGES");
+
+type Result<T> = std::result::Result<T, Box<dyn error::Error>>;
+
+struct Options {
+    out_dir: String,
+    root_dir: String,
+    force: bool,
+    no_filesystems: bool,
+    show_hardware_config: bool,
+    show_help: bool,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Error {
+    err: String,
+}
+
+fn parse_options(args: &[String]) -> Result<Options> {
+    let mut i: usize = 0;
+    let mut opts = Options {
+        out_dir: String::from("/etc/nixos"),
+        root_dir: String::from("/"),
+        force: false,
+        no_filesystems: false,
+        show_hardware_config: false,
+        show_help: false,
+    };
+
+    loop {
+        if i >= args.len() {
+            return Ok(opts);
+        }
+
+        match args[i].as_ref() {
+            "-h" | "--help" => {
+                opts.show_help = true;
+
+                return Ok(opts);
+            }
+            "--dir" => {
+                i += 1;
+
+                if i >= args.len() {
+                    bail!("'--dir' requires an argument");
+                }
+
+                opts.out_dir = args[i].clone();
+            }
+            "--root" => {
+                i += 1;
+
+                if i >= args.len() {
+                    bail!("'--root' requires an argument");
+                }
+
+                opts.root_dir = args[i].clone();
+            }
+            "--force" => {
+                opts.force = true;
+            }
+            "--no-filesystems" => {
+                opts.no_filesystems = true;
+            }
+            "--show-hardware-config" => {
+                opts.show_hardware_config = true;
+            }
+            _ => {
+                bail!("unrecognized argument '{}'", args[i]);
+            }
+        }
+
+        i += 1;
+    }
+}
+
+fn show_help() -> Result<()> {
+    let status = try_with!(
+        Command::new(MAN_EXECUTABLE)
+            .arg("nixos-generate-config")
+            .env("MANPATH", NIXOS_MANPAGES)
```

Comment:
Just call `man nixos-generate-config` here without `MANPATH`.

Reply:
This would break unit tests and adds minor impurities to `nixos-generate-config`. Are you sure about that, given that it should generate any conflict? https://github.com/NixOS/nixpkgs/pull/91353#discussion_r445590395